src/targets/gpu/fuse_mlir.cpp

```diff
@@ -314,6 +315,58 @@ void fuse_mlir::apply(module_pass_manager& mpm) const
 #endif
 }
 
+#ifdef MIGRAPHX_MLIR
+
+namespace {
+MIGRAPHX_PRED_MATCHER(is_supported_arch, instruction_ref)
+{
+    // TODO(ravil): debug
+    static std::unordered_set<std::string> supported_consumer_archs{
+        "gfx900", "gfx906", "gfx908", "gfx1030", "gfx940"};
+
+    // static std::unordered_set<std::string> supported_consumer_archs{"gfx1030"};
+    const auto device_name = trim(split_string(get_device_name(), ':').front());
+    if(contains(supported_consumer_archs, device_name))
+        return true;
+    return false;
+}
+
+struct find_mlir_standalone_convolution_op
+{
+    auto matcher() const { return match::name("convolution")(is_supported_arch); }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto conv_based_op = r.result;
+        // Only fuse with fp32/fp16
+        if(std::any_of(conv_based_op->inputs().begin(), conv_based_op->inputs().end(), [&](auto i) {
+               return not contains({shape::type_t::float_type, shape::type_t::half_type},
+                                   i->get_shape().type());
+           }))
+            return;
+
+        static size_t counter = 0;
+        module_ref mm         = mpm.create_module("mlir_" + std::to_string(counter++));
+        mm->set_bypass();
+        auto [anchor_op, top_inputs] = fuse_input_ops_and_gemm_based_op(mm, conv_based_op);
+        mm->add_return({anchor_op});
+        mpm.get_module().replace_instruction(
+            conv_based_op, mlir_op{conv_based_op->get_operator()}, top_inputs, {mm});
+    }
+};
+} // namespace
+
+#endif // MIGRAPHX_MLIR
+
+void standalone_mlir::apply(module_pass_manager& mpm) const
+{
+#ifdef MIGRAPHX_MLIR
+    match::find_matches(mpm, find_mlir_standalone_convolution_op{});
```

Comment:
This should be moved to `fuse_mlir::appy`.

Reply:
Got you. So, the I will remove
https://github.com/ravil-mobile/AMDMIGraphX/blob/63bdf8f04ab416d20023c7ac2acbb26df31172f3/src/targets/gpu/include/migraphx/gpu/standalone_mlir.hpp#L1-L47